cluster-autoscaler/cloudprovider/gce/gce_price_model_test.go

```diff
@@ -29,73 +30,118 @@ import (
-
-	// custom node
-	node3 := BuildTestNode("sillyname3", 8000, 30*units.GiB)
-	price3, err := model.NodePrice(node3, now, now.Add(time.Hour))
-	assert.NoError(t, err)
-	// custom nodes should be slightly more expensive than regular.
-	assert.True(t, price1 < price3)
-	assert.True(t, price1*1.2 > price3)
-
-	// regular with gpu
-	node4 := BuildTestNode("sillyname4", 8000, 30*units.GiB)
-	node4.Status.Capacity[gpu.ResourceNvidiaGPU] = *resource.NewQuantity(1, resource.DecimalSI)
-	node4.Labels = labels1
-	price4, _ := model.NodePrice(node4, now, now.Add(time.Hour))
-
-	// preemptible with gpu
-	node5 := BuildTestNode("sillyname5", 8000, 30*units.GiB)
-	node5.Labels = labels2
-	node5.Status.Capacity[gpu.ResourceNvidiaGPU] = *resource.NewQuantity(1, resource.DecimalSI)
-	price5, _ := model.NodePrice(node5, now, now.Add(time.Hour))
-
-	// Nodes with GPU are way more expensive than regular.
-	// Being preemptible doesn't bring much of a discount (less than 50%).
-	assert.True(t, price4 > price5)
-	assert.True(t, price4 < 1.5*price5)
-	assert.True(t, price4 > 2*price1)
-
-	// small custom node
-	node6 := BuildTestNode("sillyname6", 1000, 3750*units.MiB)
-	price6, err := model.NodePrice(node6, now, now.Add(time.Hour))
-	assert.NoError(t, err)
-	// 8 times smaller node should be 8 times less expensive.
-	assert.True(t, math.Abs(price3-8*price6) < 0.1)
+// this test is meant to cover all the branches in pricing logic, not all possible types of instances
+func TestGetNodePrice(t *testing.T) {
+	// tests assert that price(cheaperNode) < priceComparisonCoefficient * price(expensiveNode)
+	cases := map[string]struct {
+		cheaperNode                *apiv1.Node
+		expensiveNode              *apiv1.Node
+		priceComparisonCoefficient float64
+	}{
+		// instance types
+		"e2 is cheaper than n1": {
+			cheaperNode:                testNode(t, "e2", "e2-standard-8", 8000, 32*units.GiB, "", 0, false),
+			expensiveNode:              testNode(t, "n1", "n1-standard-8", 8000, 32*units.GiB, "", 0, false),
+			priceComparisonCoefficient: 1,
+		},
+		"custom nodes are more expensive than n1": {
+			cheaperNode:                testNode(t, "n1", "n1-standard-8", 8000, 32*units.GiB, "", 0, false),
+			expensiveNode:              testNode(t, "custom", "custom-8", 8000, 32*units.GiB, "", 0, false),
+			priceComparisonCoefficient: 1,
+		},
+		"custom nodes are not extremely expensive": {
+			cheaperNode:                testNode(t, "custom", "custom-8", 8000, 32*units.GiB, "", 0, false),
+			expensiveNode:              testNode(t, "n1", "n1-standard-8", 8000, 32*units.GiB, "", 0, false),
+			priceComparisonCoefficient: 1.2,
+		},
+		"custom node price scales linearly": {
+			cheaperNode:                testNode(t, "small_custom", "custom-1", 1000, 3.75*units.GiB, "", 0, false),
+			expensiveNode:              testNode(t, "large_custom", "custom-8", 8000, 32*units.GiB, "", 0, false),
+			priceComparisonCoefficient: 1.0 / 7.9,
+		},
+		"custom node price scales linearly 2": {
```

Comment:
If I understand this test case correctly, it checks that `price(custom-1) < 8.1 * price(custom-8)`. I'm guessing cheaper/expensive node were supposed to be reversed in this test case?

Reply:
Yup, thanks for catching that.